mediapipe/examples/android/src/java/com/google/mediapipe/apps/instantmotiontracking/MainActivity.java

```diff
@@ -369,47 +381,61 @@ public void onClick(View v) {
-        btn.setPadding(25,25,25,25);
-        btn.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
-    }
+  // Sets ImageButton UI for Sticker Buttons
+  public void setStickerButtonDesign(ImageButton btn, int imageDrawable) {
+      btn.setImageDrawable(getResources().getDrawable(imageDrawable));
+      btn.setBackground(getResources().getDrawable(R.drawable.circle_button));
+      btn.setLayoutParams(new LinearLayout.LayoutParams(250,250));
+      btn.setPadding(25,25,25,25);
+      btn.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
+  }
 
-public void setGIFBitmaps(String gif_url) {
-  Glide.with(this).asGif().load(gif_url).into(new SimpleTarget<GifDrawable>() {
-    @Override
-    public void onResourceReady(GifDrawable resource, Transition<? super GifDrawable> transition) {
-      try {
-        Object startConstant = resource.getConstantState();
-        Field frameManager = startConstant.getClass().getDeclaredField("frameLoader");
-        frameManager.setAccessible(true);
-        Object frameLoader = frameManager.get(startConstant);
-
-        Field decoder = frameLoader.getClass().getDeclaredField("gifDecoder");
-        decoder.setAccessible(true);
-        StandardGifDecoder GIFDecoder = (StandardGifDecoder) decoder.get(frameLoader);
-        for (int i = 0; i < GIFDecoder.getFrameCount(); i++) {
-          GIFDecoder.advance();
-          GIFBitmaps.add(GIFDecoder.getNextFrame());
+  public void setGIFBitmaps(String gif_url) {
+    Glide.with(this).asGif().load(gif_url).into(new SimpleTarget<GifDrawable>() {
+      @Override
+      public void onResourceReady(GifDrawable resource, Transition<? super GifDrawable> transition) {
+        try {
+          Object startConstant = resource.getConstantState();
+          Field frameManager = startConstant.getClass().getDeclaredField("frameLoader");
+          frameManager.setAccessible(true);
+          Object frameLoader = frameManager.get(startConstant);
+
+          Field decoder = frameLoader.getClass().getDeclaredField("gifDecoder");
+          decoder.setAccessible(true);
+          StandardGifDecoder GIFDecoder = (StandardGifDecoder) decoder.get(frameLoader);
+          for (int i = 0; i < GIFDecoder.getFrameCount(); i++) {
+            GIFDecoder.advance();
+            GIFBitmaps.add(GIFDecoder.getNextFrame());
+          }
+        }
+        catch (Exception e) {
+          e.printStackTrace();
         }
       }
-      catch (Exception e) {
-        e.printStackTrace();
-      }
+    });
+  }
+
+  // Function that is continuously called in order to time GIF frame updates
+  public void updateGIFFrame() {
+    long millisPerFrame = 1000/GIF_FRAME_RATE;
+    if(System.currentTimeMillis() - gifLastFrameUpdateMS >= millisPerFrame) {
+      // Update GIF timestamp
+      gifLastFrameUpdateMS = System.currentTimeMillis();
+      // Cycle through every possible frame
+      if(gifCurrentIndex >= GIFBitmaps.size() - 1)
```

Comment:
can just replace this logic with:
gifCurrentIndex = (gifCurrentIndex + 1) % GIFBitmaps.size();